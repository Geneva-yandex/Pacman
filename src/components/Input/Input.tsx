import * as React from 'react';
import {ChangeEvent} from 'react';

type OwnProps = {
    className?: string;
    value?: string | number;
    type: string;
    title: string;
    placeholder?: string;
    name: string;
    onChange?: (event: ChangeEvent) => void;
};

type Props = OwnProps;

export default class Input extends React.Component<Props> {
    public render() {
        return (
            <div className={this.props.className}>
                <label>
                    <span>{this.props.title}</span>
                    <input {...this.props} />
                </label>
            </div>
        );
    }
}
