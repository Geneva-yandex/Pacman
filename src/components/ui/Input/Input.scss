@import "src/styles/variables";

.Input {
  &__label {
    display: block;
    font-size: 14px;
    color: #bdbdbd;
    font-weight: 400;
    margin-bottom: 8px;
  }

  &__control {
    background: rgba(#ffffff, 5%);
    border: 1px solid rgba(#ffffff, 20%);
    height: 46px;
    border-radius: 4px;
    color: #ffffff;
    font-size: 16px;
    padding: 5px 10px;
    font-weight: 400;
    outline: 0;
    transition: 0.25s border-color ease;

    &:focus {
      border-color: rgba(#ffffff, 40%);
    }

    &::placeholder {
      color: #b8bcca;
    }
  }
}
