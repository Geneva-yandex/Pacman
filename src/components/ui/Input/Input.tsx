import React from 'react';
import {ChangeEvent, createRef} from 'react';
import bem from 'easy-bem';
import './Input.scss';
import classnames from 'classnames';

type OwnProps = {
    className?: string;
    value?: string | number;
    type: string;
    title?: string;
    placeholder?: string;
    name: string;
    ref?: any;
    onChange?: (event: ChangeEvent) => void;
};

type Props = OwnProps;

const b = bem('Input');

export default class Input extends React.PureComponent<Props> {
    private _inputRef = createRef<HTMLInputElement>();

    public get inputRef(): HTMLInputElement | null {
        return this._inputRef.current;
    }

    public render() {
        const className = classnames(b(), this.props.className);

        return (
            <div className={className}>
                <label>
                    {this.props.title && <span className={b('label')}>{this.props.title}</span>}
                    <input className={b('control')} {...this.props} ref={this._inputRef} />
                </label>
            </div>
        );
    }
}
