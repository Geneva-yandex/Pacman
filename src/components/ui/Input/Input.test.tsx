import * as React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

const props = {
    type: 'text',
    name: 'name'
};

describe('Input', () => {
    test('Input component renders correctly', () => {
        const renderedComponent = renderer
            .create(<Input {...props}/>)
            .toJSON();

        expect(renderedComponent).toMatchSnapshot();
    });
});
