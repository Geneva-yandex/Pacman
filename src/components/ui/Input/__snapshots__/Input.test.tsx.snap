// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input component renders correctly 1`] = `
<div
  className="Input"
>
  <label>
    <input
      className="Input__control"
      name="name"
      type="text"
    />
  </label>
</div>
`;
